src/Proto.Mailbox/Mailbox.cs

```diff
@@ -115,10 +115,9 @@ private Task RunAsync()
                     _stats[i].MailboxEmpty();
                 }
             }
-            return Task.FromResult(0);
         }
 
-        private bool ProcessMessages()
+        private async Task<bool> ProcessMessages()
```

Comment:
That's what happens in the `RescheduleOnTaskComplete` check for faulted that is.  Then tries to send to `EscalateFailureAsync` again with the fault from the task.

Can add some logging.